Prep 4.0 launcher improvements

kuceb opened the pull request:
- async/await-ify browser/chrome.ts
- update helper/e2e.js to allow running only on select browsers (requires patch to snap-shot-it for now)

kuceb committed:
async/await-ify chrome.ts, update e2e.js

brian-mann commented on packages/server/test/support/helpers/e2e.js:
```diff
@@ -319,7 +358,55 @@ localItFn.skip = function (title, options) {
+
+      return Promise.delay(400).then(() => retry())
+    }
+  }
```
move this logic to `cy.task()` in the spec

kuceb committed:
remove psInclude

bahmutov commented on packages/server/patches/snap-shot-it+7.9.1.patch:
```diff
@@ -0,0 +1,13 @@
+ 
+ // eslint-disable-next-line immutable/no-let
+-let pruneSnapshots
++let pruneSnapshots = function noop() {}
```
this is fixed by https://github.com/bahmutov/snap-shot-it/releases/tag/v7.9.2

brian-mann committed:
bump snap-shot-it, remove patch-package

brian-mann committed:
fix failing tests, tighten up code

brian-mann committed:
remove unused Promise.all

brian-mann committed:
more code cleanup from coffeescript conversion

kuceb committed:
fix expectedExitcode

kuceb committed:
update snapshot with cross origin error message

brian-mann committed:
more code cleanup from coffeescript conversion

brian-mann committed:
remove global state, cleanup normalizing stdout in snapshots

brian-mann committed:
fix asserting on the expected exit code

brian-mann committed:
remove firefox a default browser in e2e tests

brian-mann committed:
remove dupe const

brian-mann committed:
make onStdout return first arg by default

brian-mann committed:
make expectedExitCode: 0 the default, remove duplicate option from all e2e tests

brian-mann committed:
reuse _getArgs() properly, tighten up dupe code, move perf e2e tests to use new e2e.it helpers

brian-mann committed:
make e2e test pass properly, and capture snapshot

- this was being overlooked because expectedExitCode wasn’t being set
(until now its the default)

brian-mann committed:
revert firefox specific changes to reduce PR diff

brian-mann committed:
remove newline
